pkg/es/config/config_test.go

```diff
@@ -0,0 +1,629 @@
+		indexFrequency string
+		expected       time.Duration
+	}{
+		{
+			name:           "hourly jaeger-span",
+			indexFrequency: "hour",
+			expected:       -1 * time.Hour,
+		},
+		{
+			name:           "daily jaeger-span",
+			indexFrequency: "daily",
+			expected:       -24 * time.Hour,
+		},
+		{
+			name:           "empty jaeger-span",
+			indexFrequency: "",
+			expected:       -24 * time.Hour,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			c := &Configuration{IndexRolloverFrequencySpans: test.indexFrequency}
+			got := c.GetIndexRolloverFrequencySpansDuration()
+			require.Equal(t, test.expected, got)
+		})
+	}
+}
+
+func TestGetIndexRolloverFrequencyServicesDuration(t *testing.T) {
+	tests := []struct {
+		name           string
+		indexFrequency string
+		expected       time.Duration
+	}{
+		{
+			name:           "hourly jaeger-service",
+			indexFrequency: "hour",
+			expected:       -1 * time.Hour,
+		},
+		{
+			name:           "daily jaeger-service",
+			indexFrequency: "daily",
+			expected:       -24 * time.Hour,
+		},
+		{
+			name:           "empty jaeger-service",
+			indexFrequency: "",
+			expected:       -24 * time.Hour,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			c := &Configuration{IndexRolloverFrequencyServices: test.indexFrequency}
+			got := c.GetIndexRolloverFrequencyServicesDuration()
+			require.Equal(t, test.expected, got)
+		})
+	}
+}
+
+func TestGetIndexRolloverFrequencySamplingDuration(t *testing.T) {
+	tests := []struct {
+		name           string
+		indexFrequency string
+		expected       time.Duration
+	}{
+		{
+			name:           "hourly jaeger-sampling",
+			indexFrequency: "hour",
+			expected:       -1 * time.Hour,
+		},
+		{
+			name:           "daily jaeger-sampling",
+			indexFrequency: "daily",
+			expected:       -24 * time.Hour,
+		},
+		{
+			name:           "empty jaeger-sampling",
+			indexFrequency: "",
+			expected:       -24 * time.Hour,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			c := &Configuration{IndexRolloverFrequencySampling: test.indexFrequency}
+			got := c.GetIndexRolloverFrequencySamplingDuration()
+			require.Equal(t, test.expected, got)
+		})
+	}
+}
+
+func TestValidate(t *testing.T) {
+	tests := []struct {
+		name          string
+		config        *Configuration
+		expectedError bool
+	}{
+		{
+			name: "All valid input are set",
+			config: &Configuration{
+				Servers: []string{"localhost:8000/dummyserver"},
+			},
+			expectedError: false,
+		},
+		{
+			name:          "no valid input are set",
+			config:        &Configuration{},
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := test.config.Validate()
+			if test.expectedError {
+				require.Error(t, got)
+			} else {
+				require.NoError(t, got)
+			}
+		})
+	}
+}
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
```

Comment:
not correct - see https://github.com/jaegertracing/jaeger/blob/4220d62cf3e2f18356c8504666ff43a1d838c241/pkg/version/package_test.go